tests/certification/bindings/azure/eventhubs/eventhubs_test.go

```diff
@@ -0,0 +1,181 @@
+					Name:      "azure-eventhubs-binding",
+					Operation: "create",
+					Data:      []byte(msg),
+				})
+			require.NoError(ctx, err, "error publishing message")
+		}
+
+		// Assert the observed messages
+		consumerGroup1.Assert(ctx, time.Minute)
+		return nil
+	}
+
+	// Application logic that tracks messages from eventhub.
+	application := func(ctx flow.Context, s common.Service) (err error) {
+		// Simulate periodic errors.
+		sim := simulate.PeriodicError(ctx, 100)
+		// Setup the binding endpoints
+		err = multierr.Combine(err,
+			s.AddBindingInvocationHandler("azure-eventhubs-binding", func(_ context.Context, in *common.BindingEvent) ([]byte, error) {
+				consumerGroup1.Observe(string(in.Data))
+				if err := sim(); err != nil {
+					return nil, err
+				}
+				ctx.Logf("Receiving eventhubs message: %s", string(in.Data))
+				return []byte("{}"), nil
+			}))
+		return err
+	}
+
+	publishMessages := func(ctx flow.Context) error {
+		// Define what is expected
+		outputmsg := make([]string, numMessages)
+		for i := 0; i < numMessages; i++ {
+			outputmsg[i] = fmt.Sprintf("publish messages to device: Message %03d", i)
+		}
+		consumerGroup2.ExpectStrings(outputmsg...)
+		cmd := exec.Command("/bin/bash", "../../../tests/scripts/send-iot-device-events.sh")
+		out, err := cmd.CombinedOutput()
+		assert.Nil(t, err, "Error in send-iot-device-events.sh:\n%s", out)
+		consumerGroup2.Assert(ctx, time.Minute)
+		return nil
+	}
+
+	// Flow of events: Start app, sidecar, interrupt network to check reconnection, send and receive
+	flow.New(t, "eventhubs binding authentication using service principal").
+		Step(app.Run("app", fmt.Sprintf(":%d", appPort), application)).
+		Step(sidecar.Run("sidecar",
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			embedded.WithComponentsPath("./components/binding/serviceprincipal"),
+			runtime.WithSecretStores(secrets_components),
+			runtime.WithOutputBindings(out_component),
+			runtime.WithInputBindings(in_component),
+		)).
+		Step("interrupt network", network.InterruptNetwork(30*time.Second, nil, nil, "21092", "31092", "41092")).
+		Step("send and wait", sendAndReceive).
+		Run()
+
+	// Flow of events: Start app, sidecar, interrupt network to check reconnection, send and receive
+	flow.New(t, "eventhubs binding authentication using connection string").
+		Step(app.Run("app", fmt.Sprintf(":%d", appPort), application)).
+		Step(sidecar.Run("sidecar",
+			embedded.WithAppProtocol(runtime.HTTPProtocol, appPort),
+			embedded.WithDaprGRPCPort(grpcPort),
+			embedded.WithDaprHTTPPort(httpPort),
+			embedded.WithComponentsPath("./components/binding/connectionstring"),
+			runtime.WithSecretStores(secrets_components),
+			runtime.WithOutputBindings(out_component),
+			runtime.WithInputBindings(in_component),
+		)).
+		Step("interrupt network", network.InterruptNetwork(30*time.Second, nil, nil, "21092", "31092", "41092")).
```

Comment:
What are these ports? This does not seem to be the ports that EventHubs could be using.